File: Algorithms/yandex_handbook_algo/Chapter_3/3.6-Randomizing/taskA.cpp
#include <vector>
#include <algorithm>
#include <iostream>

void Lomuto_partition(std::vector<int>& data, int pivot)
{
    int i = 0;
    for (int j = 1; j < data.size(); j++)
    {
        if (data[j] <= pivot)
        {
            i++;
            std::swap(data[i], data[j]);
        }
    }
   std::swap(data[i], data[0]);
    
}

int main()
{
    int n, p;
    std::cin >> n;
    std::vector<int> data;
    for (int j = 0; j < n; ++j)
    {
        std::cin >> p;
        data.push_back(p);
    }
    Lomuto_partition(data, data[0]);
    for (int j = 0; j < n; ++j)
    {
        std::cout << data[j] << ' ';
    }
    return 0;
}

File: Algorithms/yandex_handbook_algo/3.6-Randomizing/taskA.py
def Lomuto_partition(data, pivot):
    i = 0
    for j in range(1, len(data)):
        if data[j] <= pivot:
            i += 1
            data[i], data[j] = data[j], data[i]
    data[0], data[i] = data[i], data[0]
    return data


n = int(input())
data = [int(x) for x in input().split()]
data = Lomuto_partition(data, data[0])
print(' '.join(map(str, data)))